crates/worker/src/partition/mod.rs

```diff
@@ -417,64 +456,71 @@ where
                     // clear buffers used when applying the next record
                     action_collector.clear();
 
-                    for (lsn, envelope) in command_buffer.drain(..) {
-                        let command_start = Instant::now();
-
-                        trace!(%lsn, "Processing bifrost record for '{}': {:?}", envelope.command.name(), envelope.header);
-
-                        let leadership_change = self.apply_record(
-                            lsn,
-                            envelope,
-                            &mut transaction,
-                            &mut action_collector).await?;
-
-                        apply_command_latency.record(command_start.elapsed());
-
-                        if let Some((header, announce_leader)) = leadership_change {
-                            // commit all changes so far, this is important so that the actuators see all changes
-                            // when becoming leader.
-                            transaction.commit().await?;
-
-                            // We can ignore all actions collected so far because as a new leader we have to instruct the
-                            // actuators afresh.
-                            action_collector.clear();
-
-                            self.status.last_observed_leader_epoch = Some(announce_leader.leader_epoch);
-                            if header.source.is_processor_generational() {
-                                let Source::Processor { generational_node_id, .. } = header.source else {
-                                    unreachable!("processor source must have generational_node_id");
-                                };
-                                // all new AnnounceLeader messages should come from a PartitionProcessor
-                                self.status.last_observed_leader_node = generational_node_id;
-                            } else if announce_leader.node_id.is_some() {
-                                // older AnnounceLeader messages have the announce_leader.node_id set
-                                self.status.last_observed_leader_node = announce_leader.node_id;
-                            }
+                    for record in command_buffer.drain(..) {
+                        match record {
+                            Record::Envelope(lsn, envelope) => {
```

Comment:
The `Envelope` handling logic is unchanged, this block is just indented due to the match expression needed to handle gaps.